tests/main/snap-auto-import-asserts-spools/task.yaml

```diff
@@ -0,0 +1,43 @@
+summary: Check that `snap auto-import` works as expected
+
+systems: [ubuntu-core-16-64]
+
+prepare: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+    echo "Ensure the testrootorg-store.account-key is not already added"
+    output=$(snap known account-key | grep "name: test-store"|wc -l)
+    if [ "$output" != "0" ]; then
+            echo " testrootorg-store.account-key is already added"
+            exit 1
+    fi
+    echo "Create a ramdisk with the testrootorg-store.account-key assertion"
+    mkfs.vfat /dev/ram0
+    mount /dev/ram0 /mnt
+    cp $TESTSLIB/assertions/testrootorg-store.account-key /mnt/auto-import.assert
+
+restore: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+execute: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+    echo "Simulate a not running snapd (happens on e.g. early boot)"
+    systemctl stop snapd.service snapd.socket
+
+    echo "`snap auto-import` spools assertions if it can not talk to snapd"
+    snap auto-import
+    ls /var/lib/snapd/spool
+    umount /mnt
+    systemctl start snapd.service snapd.socket
+
+    echo "`snap auto-import` reads from the spool dir"
+    snap auto-import
```

Comment:
This PR looks good. Do we need another one to run auto-import automatically somewhere on boot after snapd runs?